ambry-store/src/main/java/com.github.ambry.store/Log.java

```diff
@@ -34,13 +39,15 @@
 class Log implements Write {
   private final String dataDir;
   private final long capacityInBytes;
+  private final boolean isLogSegmented;
   private final StoreMetrics metrics;
+  private final Iterator<Pair<String, String>> segmentNameAndFileNameIterator;
```

Comment:
How does this data structure differ from segmentsByName ? Can these two be out of sync at any time?